src/topology/algebra/infinite_sum.lean

```diff
@@ -714,196 +730,312 @@ end encodable
+  ∏' x : (⋃ i ∈ s, t i), f x = ∏ i in s, ∏' x : t i, f x :=
+(has_prod_prod_disjoint _ hd (λ i hi, (hf i hi).has_prod)).tprod_eq
 
-lemma tsum_even_add_odd {f : ℕ → α} (he : summable (λ k, f (2 * k)))
-  (ho : summable (λ k, f (2 * k + 1))) :
-  (∑' k, f (2 * k)) + (∑' k, f (2 * k + 1)) = ∑' k, f k :=
-(he.has_sum.even_add_odd ho.has_sum).tsum_eq.symm
+@[to_additive tsum_even_add_odd]
+lemma tprod_even_mul_odd {f : ℕ → α} (he : prodable (λ k, f (2 * k)))
+  (ho : prodable (λ k, f (2 * k + 1))) :
+  (∏' k, f (2 * k)) * (∏' k, f (2 * k + 1)) = ∏' k, f k :=
+(he.has_prod.even_mul_odd ho.has_prod).tprod_eq.symm
 
-end tsum
+end tprod
 
-section prod
+section has_continuous_star
+variables [add_comm_monoid α] [topological_space α] [star_add_monoid α] [has_continuous_star α]
+  [t2_space α] {f : β → α}
 
-variables [add_comm_monoid α] [topological_space α] [add_comm_monoid γ] [topological_space γ]
+lemma tsum_star : star (∑' b, f b) = ∑' b, star (f b) :=
+begin
+  by_cases hf : summable f,
+  { exact hf.has_sum.star.tsum_eq.symm },
+  { rw [tsum_eq_zero_of_not_summable hf, tsum_eq_zero_of_not_summable (mt summable.of_star hf),
+      star_zero] }
+end
 
-lemma has_sum.prod_mk {f : β → α} {g : β → γ} {a : α} {b : γ}
-  (hf : has_sum f a) (hg : has_sum g b) :
-  has_sum (λ x, (⟨f x, g x⟩ : α × γ)) ⟨a, b⟩ :=
-by simp [has_sum, ← prod_mk_sum, filter.tendsto.prod_mk_nhds hf hg]
+end has_continuous_star
+
+section prod
+variables [comm_monoid α] [topological_space α] [comm_monoid γ] [topological_space γ]
+
+@[to_additive has_sum.prod_mk]
+lemma has_prod.prod_mk {f : β → α} {g : β → γ} {a : α} {b : γ}
+  (hf : has_prod f a) (hg : has_prod g b) :
+  has_prod (λ x, (⟨f x, g x⟩ : α × γ)) ⟨a, b⟩ :=
+by simp [has_prod, ← prod_mk_prod, hf.prod_mk_nhds hg]
 
 end prod
 
 section pi
-variables {ι : Type*} {π : α → Type*} [∀ x, add_comm_monoid (π x)] [∀ x, topological_space (π x)]
+variables {ι : Type*} {π : α → Type*} [∀ x, comm_monoid (π x)] [∀ x, topological_space (π x)]
+  {f : ι → ∀ x, π x}
 
-lemma pi.has_sum {f : ι → ∀ x, π x} {g : ∀ x, π x} :
-  has_sum f g ↔ ∀ x, has_sum (λ i, f i x) (g x) :=
-by simp only [has_sum, tendsto_pi_nhds, sum_apply]
+@[to_additive pi.has_sum]
+lemma pi.has_prod {g : ∀ x, π x} : has_prod f g ↔ ∀ x, has_prod (λ i, f i x) (g x) :=
+by simp only [has_prod, tendsto_pi_nhds, prod_apply]
 
-lemma pi.summable {f : ι → ∀ x, π x} : summable f ↔ ∀ x, summable (λ i, f i x) :=
-by simp only [summable, pi.has_sum, skolem]
+@[to_additive pi.summable] lemma pi.prodable : prodable f ↔ ∀ x, prodable (λ i, f i x) :=
+by simp only [prodable, pi.has_prod, skolem]
 
-lemma tsum_apply [∀ x, t2_space (π x)] {f : ι → ∀ x, π x}{x : α} (hf : summable f) :
-  (∑' i, f i) x = ∑' i, f i x :=
-(pi.has_sum.mp hf.has_sum x).tsum_eq.symm
+@[to_additive tsum_apply]
+lemma tprod_apply [∀ x, t2_space (π x)] {x : α} (hf : prodable f) : (∏' i, f i) x = ∏' i, f i x :=
+(pi.has_prod.1 hf.has_prod x).tprod_eq.symm
 
 end pi
 
+section mul_opposite
+open mul_opposite
```

Comment:
I think for the sake of review it would make sense to split this move to its own PR.

Reply:
#18414

Reply:
Thanks for doing this, it's much easier to follow now.